firmware/application/apps/ble_app.cpp

```diff
@@ -0,0 +1,403 @@
+                }
+            }
+        }
+    }
+}
+
+void BleRecentEntryDetailView::set_entry(const BleRecentEntry& entry) {
+    entry_ = entry;
+    set_dirty();
+}
+
+static std::uint64_t get_freq_by_channel_number(uint8_t channel_number) {
+    uint64_t freq_hz;
+
+    switch (channel_number) {
+        case 37:
+            freq_hz = 2402000000ull;
+            break;
+        case 38:
+            freq_hz = 2426000000ull;
+            break;
+        case 39:
+            freq_hz = 2480000000ull;
+            break;
+        case 0 ... 10:
+            freq_hz = 2404000000ull + channel_number * 2000000ull;
+            break;
+        case 11 ... 36:
+            freq_hz = 2428000000ull + (channel_number - 11) * 2000000ull;
+            break;
+        default:
+            freq_hz = 0xffffffffffffffff;
+    }
+
+    return freq_hz;
+}
+
+void BLERxView::focus() {
+    field_frequency.focus();
+}
+
+BLERxView::BLERxView(NavigationView& nav)
+    : nav_{nav} {
+    baseband::run_image(portapack::spi_flash::image_tag_btle_rx);
+
+    add_children({ &rssi,
+                       &channel,
+                       &field_rf_amp,
+                       &field_lna,
+                       &field_vga,
+                       &options_region,
+                       &field_frequency,
+                       &check_log,
+#if USE_CONSOLE
+                       &console });
+#else
+                       &recent_entries_view,
+                       &recent_entry_detail_view });
+#endif
+
+    recent_entry_detail_view.hidden(true);
+
+    recent_entries_view.on_select = [this](const BleRecentEntry& entry) {
+        nav_.push<BleRecentEntryDetailView>(entry);
+    };
+
+    // field_frequency.set_value(get_freq_by_channel_number(37));
+    field_frequency.set_step(2000000);
+
+    check_log.set_value(logging);
+
+    check_log.on_select = [this](Checkbox&, bool v) {
+        str_log = "";
+        logging = v;
+    };
+
+    options_region.on_change = [this](size_t, int32_t i) {
+        if (i == 0) {
+            field_frequency.set_value(get_freq_by_channel_number(37));
+            channel_number = 37;
+        } else if (i == 1) {
+            field_frequency.set_value(get_freq_by_channel_number(38));
+            channel_number = 38;
+        } else if (i == 2) {
+            field_frequency.set_value(get_freq_by_channel_number(39));
+            channel_number = 39;
+        }
+
+        baseband::set_btle(channel_number);
+    };
+
+    options_region.set_selected_index(0, true);
+
+    logger = std::make_unique<BLELogger>();
+
+    if (logger)
+        logger->append(LOG_ROOT_DIR "/BLELOG_" + to_string_timestamp(rtc_time::now()) + ".TXT");
+
+    // Auto-configure modem for LCR RX (will be removed later)
+    baseband::set_btle(channel_number);
+
+    receiver_model.enable();
+}
+
+void BLERxView::on_data(BlePacketData* packet) {
+    std::string str_console = "";
```

Comment:
nit: You don't need to init strings like that. Their default constructor will run.
Also, if you have a guess about how long this string will be, call reserve(size_t) so that you can pre-allocate a large block.

Memory fragmentation is a real killer on this little thing so the fewer allocations the better.

Reply:
Oh I've seen this everywhere else, so I figured it was to clear the buffer each time we went through the loop.